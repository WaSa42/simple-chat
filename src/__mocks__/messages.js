import avatar1 from '../assets/images/avatar1.jpg';
import avatar2 from '../assets/images/avatar2.png';

export default {
  fr: [
    {
      _id: 'randomlyGeneratedMessageId1',
      sentAt: '1550926394001',
      content: "Tu as vu l'épisode final de How I Met Your Mother ?",
      user: {
        _id: 'randomlyGeneratedUserId1',
        userName: 'Nicolas',
        profile: {
          picture: avatar1,
        },
      },
    },
    {
      _id: 'randomlyGeneratedMessageId2',
      sentAt: '1550926522171',
      content: "Oui, c'était pas mal.",
      user: {
        _id: 'randomlyGeneratedUserId2',
        userName: 'Samy',
        profile: {
          picture: avatar2,
        },
      },
    },
    {
      _id: 'randomlyGeneratedMessageId3',
      sentAt: '1550926609209',
      content: "J'ai bien aimé moi aussi, mais il est vivement critiqué apparemment...",
      user: {
        _id: 'randomlyGeneratedUserId1',
        userName: 'Nicolas',
        profile: {
          picture: avatar1,
        },
      },
    },
    {
      _id: 'randomlyGeneratedMessageId4',
      sentAt: '1550926681241',
      content: "C'est parce que les fan ne s'attendaient pas à ça et je peux les comprendre en soit.",
      user: {
        _id: 'randomlyGeneratedUserId2',
        userName: 'Samy',
        profile: {
          picture: avatar2,
        },
      },
    },
    {
      _id: 'randomlyGeneratedMessageId5',
      sentAt: '1550926724177',
      content: 'Mais pour moi ça reste un bonne fin quand même. Après tant de saison et de blagues qui nous ont bien fait rire.',
      user: {
        _id: 'randomlyGeneratedUserId2',
        userName: 'Samy',
        profile: {
          picture: avatar2,
        },
      },
    },
    {
      _id: 'randomlyGeneratedMessageId6',
      sentAt: '1550926800577',
      content: "Carrémment ! C'est une série : Legen -",
      user: {
        _id: 'randomlyGeneratedUserId1',
        userName: 'Nicolas',
        profile: {
          picture: avatar1,
        },
      },
    },
    {
      _id: 'randomlyGeneratedMessageId7',
      sentAt: '1550926848921',
      content: '- dary ! Ah ah',
      user: {
        _id: 'randomlyGeneratedUserId2',
        userName: 'Samy',
        profile: {
          picture: avatar2,
        },
      },
    },
    {
      _id: 'randomlyGeneratedMessageId8',
      sentAt: '1550926916649',
      content: 'Bref, je dois y aller, on se parlera de série un prochaine fois.',
      user: {
        _id: 'randomlyGeneratedUserId2',
        userName: 'Samy',
        profile: {
          picture: avatar2,
        },
      },
    },
    {
      _id: 'randomlyGeneratedMessageId9',
      sentAt: '1550926916670',
      content: 'Tu me diras ce que tu penses de Bojack Horseman sur Netflix. A la prochaine.',
      user: {
        _id: 'randomlyGeneratedUserId2',
        userName: 'Samy',
        profile: {
          picture: avatar2,
        },
      },
    },
    {
      _id: 'randomlyGeneratedMessageId10',
      sentAt: '1550926923185',
      content: "Avec plaisir, c'esr une de mes séries préférées !",
      user: {
        _id: 'randomlyGeneratedUserId1',
        userName: 'Nicolas',
        profile: {
          picture: avatar1,
        },
      },
    },
    {
      _id: 'randomlyGeneratedMessageId11',
      sentAt: '1550926923190',
      content: 'Ciao ciao',
      user: {
        _id: 'randomlyGeneratedUserId1',
        userName: 'Nicolas',
        profile: {
          picture: avatar1,
        },
      },
    },
  ],
  en: [
    {
      _id: 'randomlyGeneratedMessageId1',
      sentAt: '1550926394001',
      content: 'Hey ! did you seen the season 9 final of How I Met Your Mother ?',
      user: {
        _id: 'randomlyGeneratedUserId1',
        userName: 'Nicolas',
        profile: {
          picture: avatar1,
        },
      },
    },
    {
      _id: 'randomlyGeneratedMessageId2',
      sentAt: '1550926522171',
      content: 'Oh yes, that was good.',
      user: {
        _id: 'randomlyGeneratedUserId2',
        userName: 'Samy',
        profile: {
          picture: avatar2,
        },
      },
    },
    {
      _id: 'randomlyGeneratedMessageId3',
      sentAt: '1550926609209',
      content: "I liked it too, but it's highly critized by others. ^^",
      user: {
        _id: 'randomlyGeneratedUserId1',
        userName: 'Nicolas',
        profile: {
          picture: avatar1,
        },
      },
    },
    {
      _id: 'randomlyGeneratedMessageId4',
      sentAt: '1550926681241',
      content: "That's because fans didn't expected this ending and even if I liked it, I can't blame them for that either.",
      user: {
        _id: 'randomlyGeneratedUserId2',
        userName: 'Samy',
        profile: {
          picture: avatar2,
        },
      },
    },
    {
      _id: 'randomlyGeneratedMessageId5',
      sentAt: '1550926724177',
      content: "It's still a good ending for me, after all those season you know, all those jokes and laughs. :-)",
      user: {
        _id: 'randomlyGeneratedUserId2',
        userName: 'Samy',
        profile: {
          picture: avatar2,
        },
      },
    },
    {
      _id: 'randomlyGeneratedMessageId6',
      sentAt: '1550926800577',
      content: 'Hell yeah ! We can say that the show is : Legen -',
      user: {
        _id: 'randomlyGeneratedUserId1',
        userName: 'Nicolas',
        profile: {
          picture: avatar1,
        },
      },
    },
    {
      _id: 'randomlyGeneratedMessageId7',
      sentAt: '1550926848921',
      content: '- dary ! Ah ah',
      user: {
        _id: 'randomlyGeneratedUserId2',
        userName: 'Samy',
        profile: {
          picture: avatar2,
        },
      },
    },
    {
      _id: 'randomlyGeneratedMessageId8',
      sentAt: '1550926916649',
      content: "Well, I've got to go. Should talk about tv shows another time.",
      user: {
        _id: 'randomlyGeneratedUserId2',
        userName: 'Samy',
        profile: {
          picture: avatar2,
        },
      },
    },
    {
      _id: 'randomlyGeneratedMessageId9',
      sentAt: '1550926916670',
      content: "You'll tell me what you think about Bojack Horseman. :-) See ya'",
      user: {
        _id: 'randomlyGeneratedUserId2',
        userName: 'Samy',
        profile: {
          picture: avatar2,
        },
      },
    },
    {
      _id: 'randomlyGeneratedMessageId10',
      sentAt: '1550926923185',
      content: 'With pleasure, this is one of my favorite show.',
      user: {
        _id: 'randomlyGeneratedUserId1',
        userName: 'Nicolas',
        profile: {
          picture: avatar1,
        },
      },
    },
    {
      _id: 'randomlyGeneratedMessageId11',
      sentAt: '1550926923190',
      content: 'Ciao',
      user: {
        _id: 'randomlyGeneratedUserId1',
        userName: 'Nicolas',
        profile: {
          picture: avatar1,
        },
      },
    },
  ],
};
