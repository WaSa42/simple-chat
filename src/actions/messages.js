import { normalize } from 'normalizr';
import { messageListSchema, messageSchema } from '../schemas/message';
import { receiveEntities } from './entities';

export const MESSAGES_RESET = 'MESSAGES_RESET';
export const MESSAGES_ADD = 'MESSAGES_ADD';
export const MESSAGE_SENT = 'MESSAGE_SENT';

export function resetMessages() {
  return { type: MESSAGES_RESET };
}

export function addMessages(response) {
  const { messages, totalMessages, skip, limit } = response;

  return (dispatch) => {
    const normalized = normalize(messages, messageListSchema);
    const { result, entities } = normalized;

    dispatch(receiveEntities(entities));
    dispatch({ type: MESSAGES_ADD, ids: result, totalMessages, skip, limit });
  };
}

export function addSentMessage(response) {
  const { message } = response;

  return (dispatch) => {
    const normalized = normalize(message, messageSchema);
    const { result, entities } = normalized;

    dispatch(receiveEntities(entities));
    dispatch({ type: MESSAGE_SENT, id: result });
  };
}

export function updateMessage(response) {
  const { message } = response;

  return (dispatch) => {
    const normalized = normalize(message, messageSchema);
    const { entities } = normalized;

    dispatch(receiveEntities(entities));
  };
}
