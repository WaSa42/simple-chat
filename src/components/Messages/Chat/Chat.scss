@import '../../../theme.scss';

.chat {
  position: relative;
  margin-top: -1rem;

  .chat-window {
    overflow: hidden;
    padding-bottom: 71px;
    height: calc(100vh - 62px + 0.5rem);

    .messages {
      overflow-y: scroll;
      padding-right: 0;
      height: 100%;

      &::-webkit-scrollbar {
        display: none;
      }
    }

    .fetch-more {
      .btn-link {
        color: $primary;
        &[disabled] {
          color: inherit;
          text-decoration: none;
          cursor: auto;
        }
      }
    }

    .bottom-wrapper {
      position: absolute;
      bottom: 0;
      left: 0;
      padding: 1rem 0;
      width: 100%;
      border-top: 1px solid #d6d8db;
    }

    @media (min-width: 768px) {
      height: calc(100vh - 62px);
    }
  }
}
