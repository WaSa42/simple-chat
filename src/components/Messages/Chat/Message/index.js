import React from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';
import className from 'classnames';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faEye } from '@fortawesome/free-solid-svg-icons/faEye';
import { faEyeSlash } from '@fortawesome/free-solid-svg-icons/faEyeSlash';

import './Message.scss';

const Message = ({ content, isHidden, isUser, onHide, picture, sentAt, userName }) => (
  <li className={className('message', { left: !isUser, right: isUser, hidden: isHidden })}>
    <img className="avatar" src={picture} alt="Avatar" />
    <div className="text-wrapper">
      <div className="text">
        <div className="info">
          <span className="mr-2">{userName}</span>
          <time className="d-none d-sm-inline">{moment.unix(sentAt / 1000).format('LLL')}</time>
        </div>
        {content}
        {(onHide && isUser) && (
          <button onClick={onHide} className="btn-hide btn-nale">
            <FontAwesomeIcon icon={isHidden ? faEye : faEyeSlash} />
          </button>
        )}
      </div>
    </div>
  </li>
);

Message.propTypes = {
  content: PropTypes.string.isRequired,
  isHidden: PropTypes.bool,
  isUser: PropTypes.bool.isRequired,
  onHide: PropTypes.func,
  picture: PropTypes.string.isRequired,
  sentAt: PropTypes.string.isRequired,
  userName: PropTypes.string.isRequired,
};

Message.defaultProps = {
  isHidden: false,
  onHide: null,
};

export default Message;
