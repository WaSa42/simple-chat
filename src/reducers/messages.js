import { createReducer } from './_utilities';
import { MESSAGES_RESET, MESSAGES_ADD } from '../actions/messages';

const initialState = {
  ids: [],
  limit: 10,
  skip: 0,
  totalMessages: 0,
  updatedAt: null,
};

function resetMessages() {
  return initialState;
}

function addMessages(state, { ids, limit, skip, totalMessages }) {
  return {
    ...state,
    ids: Array.from(new Set(state.ids.concat(ids))),
    limit,
    needRefresh: false,
    skip,
    totalMessages,
    updatedAt: Date.now(),
  };
}

export default createReducer(initialState, {
  [MESSAGES_RESET]: resetMessages,
  [MESSAGES_ADD]: addMessages,
});
