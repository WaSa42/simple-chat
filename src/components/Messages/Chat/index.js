import React from 'react';
import PropTypes from 'prop-types';
import orderBy from 'lodash/orderBy';
import isEmpty from 'lodash/isEmpty';
import { translate } from 'react-i18next';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSpinner, faPaperPlane } from '@fortawesome/free-solid-svg-icons';

import './Chat.scss';
import Message from './Message';

const $ = window.jQuery;

class Chat extends React.Component {
  constructor(props) {
    super(props);
    this.state = { content: props.inputValue, wasAtBottom: false };
  }

  componentDidMount() {
    this.scrollToBottom();
  }

  componentWillReceiveProps(nextProps) {
    const { inputValue } = this.props;
    if (nextProps.inputValue !== inputValue) { this.setState({ content: nextProps.inputValue }); }

    const element = $(this.messages);
    this.setState({ wasAtBottom: element[0].scrollHeight - element.scrollTop() === element.outerHeight() });
  }

  componentDidUpdate() {
    const { wasAtBottom } = this.state;
    if (wasAtBottom) {
      this.scrollToBottom();
    }
  }

  scrollToBottom() {
    this.messagesEnd.scrollIntoView({/* behavior: 'smooth' */});
  }

  handleFetchMore(e) {
    e.preventDefault();
    const { isFetching, onFetchMore } = this.props;

    if (!isFetching) {
      onFetchMore();
    }

    return false;
  }

  handleSubmit(e) {
    e.preventDefault();
    const { isSending, onSubmit } = this.props;
    const { content } = this.state;

    if (!isSending && !isEmpty(content)) {
      onSubmit({ message: content });
      this.setState({ content: '' });
    }

    return false;
  }

  renderMessages() {
    const { messages, userId } = this.props;

    return orderBy(messages, ['sentAt'], ['asc']).map(message => (
      <Message
        {...message}
        isUser={message.user._id === userId}
        key={message._id}
        picture={message.user.profile.picture}
        sentAt={message.sentAt}
        userName={message.user.userName}
      />
    ));
  }

  render() {
    const { children, isFetching, isSending, remainMessages, t, totalMessages } = this.props;
    const { content } = this.state;

    return (
      <div id="chat" className="chat">
        <div className="chat-window">
          <ul className="messages pt-3" ref={(el) => { this.messages = el; }}>
            {totalMessages > 10 && (
              <li className="fetch-more text-center mb-3">
                <button
                  className="btn btn-link"
                  disabled={!remainMessages || isFetching}
                  onClick={e => this.handleFetchMore(e)}
                  onKeyPress={e => this.handleFetchMore(e)}
                >
                  {isFetching && <FontAwesomeIcon icon={faSpinner} spin={isFetching} />}
                  <span className="ml-2">
                    {t(`component:Chat.${remainMessages ? 'fetchMore' : 'noMoreMessages'}`)}
                  </span>
                </button>
              </li>
            )}
            {this.renderMessages()}
            <div ref={(el) => { this.messagesEnd = el; }} />
          </ul>
          {children}
          <form
            className="bottom-wrapper clearfix"
            onSubmit={e => this.handleSubmit(e)}
          >
            <div className="input-group">
              <input
                autoComplete="off"
                className="form-control"
                name="content"
                onChange={e => this.setState({ content: e.target.value })}
                placeholder={t('component:Chat.placeholder')}
                value={content}
              />
              <div className="input-group-append">
                <button disabled={isSending} type="submit" className="btn btn-default">
                  <FontAwesomeIcon icon={isSending ? faSpinner : faPaperPlane} spin={isSending} />
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    );
  }
}

Chat.propTypes = {
  children: PropTypes.element,
  isFetching: PropTypes.bool.isRequired,
  isSending: PropTypes.bool.isRequired,
  inputValue: PropTypes.string,
  messages: PropTypes.arrayOf(PropTypes.shape({
    _id: PropTypes.string.isRequired,
    content: PropTypes.string.isRequired,
    sentAt: PropTypes.string.isRequired,
    user: PropTypes.shape({
      profile: PropTypes.shape({ picture: PropTypes.string.isRequired }).isRequired,
      userName: PropTypes.string.isRequired,
    }),
  })),
  onFetchMore: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  remainMessages: PropTypes.bool,
  t: PropTypes.func.isRequired,
  totalMessages: PropTypes.number,
  userId: PropTypes.string.isRequired,
};

Chat.defaultProps = {
  children: null,
  inputValue: '',
  messages: [],
  remainMessages: false,
  totalMessages: 0,
};

export default translate()(Chat);
