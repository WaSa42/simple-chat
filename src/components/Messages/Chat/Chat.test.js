import React from 'react';
import renderer from 'react-test-renderer';

import messages from '../../../__mocks__/messages';
import Chat from './index';

jest.mock('react-i18next', () => ({
  // this mock makes sure any components using the translate HoC receive the t function as a prop
  translate: () => (Component) => {
    const HoCResponse = Component;
    HoCResponse.defaultProps = { ...Component.defaultProps, t: k => k };
    return HoCResponse;
  },
}));

it('renders correctly', () => {
  const tree = renderer.create(<Chat
    isFetching={false}
    isSending={false}
    messages={messages.fr}
    onFetchMore={() => console.log('onFetchMore')}
    onSubmit={values => console.log('onSubmit', values)}
    remainMessages={false}
    totalMessages={messages.fr.length}
    userId="randomlyGeneratedMessageId1"
  />).toJSON();

  expect(tree).toMatchSnapshot();
});
