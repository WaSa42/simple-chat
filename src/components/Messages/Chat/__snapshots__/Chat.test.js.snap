// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="chat"
  id="chat"
>
  <div
    className="chat-window"
  >
    <ul
      className="messages pt-3"
    >
      <li
        className="fetch-more text-center mb-3"
      >
        <button
          className="btn btn-link"
          disabled={true}
          onClick={[Function]}
          onKeyPress={[Function]}
        >
          <span
            className="ml-2"
          >
            component:Chat.noMoreMessages
          </span>
        </button>
      </li>
      <li
        className="message left"
      >
        <img
          alt="Avatar"
          className="avatar"
          src="avatar1.jpg"
        />
        <div
          className="text-wrapper"
        >
          <div
            className="text"
          >
            <div
              className="info"
            >
              <span
                className="mr-2"
              >
                Nicolas
              </span>
              <time
                className="d-none d-sm-inline"
              >
                February 23, 2019 1:53 PM
              </time>
            </div>
            Tu as vu l'épisode final de How I Met Your Mother ?
          </div>
        </div>
      </li>
      <li
        className="message left"
      >
        <img
          alt="Avatar"
          className="avatar"
          src="avatar2.png"
        />
        <div
          className="text-wrapper"
        >
          <div
            className="text"
          >
            <div
              className="info"
            >
              <span
                className="mr-2"
              >
                Samy
              </span>
              <time
                className="d-none d-sm-inline"
              >
                February 23, 2019 1:55 PM
              </time>
            </div>
            Oui, c'était pas mal.
          </div>
        </div>
      </li>
      <li
        className="message left"
      >
        <img
          alt="Avatar"
          className="avatar"
          src="avatar1.jpg"
        />
        <div
          className="text-wrapper"
        >
          <div
            className="text"
          >
            <div
              className="info"
            >
              <span
                className="mr-2"
              >
                Nicolas
              </span>
              <time
                className="d-none d-sm-inline"
              >
                February 23, 2019 1:56 PM
              </time>
            </div>
            J'ai bien aimé moi aussi, mais il est vivement critiqué apparemment...
          </div>
        </div>
      </li>
      <li
        className="message left"
      >
        <img
          alt="Avatar"
          className="avatar"
          src="avatar2.png"
        />
        <div
          className="text-wrapper"
        >
          <div
            className="text"
          >
            <div
              className="info"
            >
              <span
                className="mr-2"
              >
                Samy
              </span>
              <time
                className="d-none d-sm-inline"
              >
                February 23, 2019 1:58 PM
              </time>
            </div>
            C'est parce que les fan ne s'attendaient pas à ça et je peux les comprendre en soit.
          </div>
        </div>
      </li>
      <li
        className="message left"
      >
        <img
          alt="Avatar"
          className="avatar"
          src="avatar2.png"
        />
        <div
          className="text-wrapper"
        >
          <div
            className="text"
          >
            <div
              className="info"
            >
              <span
                className="mr-2"
              >
                Samy
              </span>
              <time
                className="d-none d-sm-inline"
              >
                February 23, 2019 1:58 PM
              </time>
            </div>
            Mais pour moi ça reste un bonne fin quand même. Après tant de saison et de blagues qui nous ont bien fait rire.
          </div>
        </div>
      </li>
      <li
        className="message left"
      >
        <img
          alt="Avatar"
          className="avatar"
          src="avatar1.jpg"
        />
        <div
          className="text-wrapper"
        >
          <div
            className="text"
          >
            <div
              className="info"
            >
              <span
                className="mr-2"
              >
                Nicolas
              </span>
              <time
                className="d-none d-sm-inline"
              >
                February 23, 2019 2:00 PM
              </time>
            </div>
            Carrémment ! C'est une série : Legen -
          </div>
        </div>
      </li>
      <li
        className="message left"
      >
        <img
          alt="Avatar"
          className="avatar"
          src="avatar2.png"
        />
        <div
          className="text-wrapper"
        >
          <div
            className="text"
          >
            <div
              className="info"
            >
              <span
                className="mr-2"
              >
                Samy
              </span>
              <time
                className="d-none d-sm-inline"
              >
                February 23, 2019 2:00 PM
              </time>
            </div>
            - dary ! Ah ah
          </div>
        </div>
      </li>
      <li
        className="message left"
      >
        <img
          alt="Avatar"
          className="avatar"
          src="avatar2.png"
        />
        <div
          className="text-wrapper"
        >
          <div
            className="text"
          >
            <div
              className="info"
            >
              <span
                className="mr-2"
              >
                Samy
              </span>
              <time
                className="d-none d-sm-inline"
              >
                February 23, 2019 2:01 PM
              </time>
            </div>
            Bref, je dois y aller, on se parlera de série un prochaine fois.
          </div>
        </div>
      </li>
      <li
        className="message left"
      >
        <img
          alt="Avatar"
          className="avatar"
          src="avatar2.png"
        />
        <div
          className="text-wrapper"
        >
          <div
            className="text"
          >
            <div
              className="info"
            >
              <span
                className="mr-2"
              >
                Samy
              </span>
              <time
                className="d-none d-sm-inline"
              >
                February 23, 2019 2:01 PM
              </time>
            </div>
            Tu me diras ce que tu penses de Bojack Horseman sur Netflix. A la prochaine.
          </div>
        </div>
      </li>
      <li
        className="message left"
      >
        <img
          alt="Avatar"
          className="avatar"
          src="avatar1.jpg"
        />
        <div
          className="text-wrapper"
        >
          <div
            className="text"
          >
            <div
              className="info"
            >
              <span
                className="mr-2"
              >
                Nicolas
              </span>
              <time
                className="d-none d-sm-inline"
              >
                February 23, 2019 2:02 PM
              </time>
            </div>
            Avec plaisir, c'esr une de mes séries préférées !
          </div>
        </div>
      </li>
      <li
        className="message left"
      >
        <img
          alt="Avatar"
          className="avatar"
          src="avatar1.jpg"
        />
        <div
          className="text-wrapper"
        >
          <div
            className="text"
          >
            <div
              className="info"
            >
              <span
                className="mr-2"
              >
                Nicolas
              </span>
              <time
                className="d-none d-sm-inline"
              >
                February 23, 2019 2:02 PM
              </time>
            </div>
            Ciao ciao
          </div>
        </div>
      </li>
      <div />
    </ul>
    <form
      className="bottom-wrapper clearfix"
      onSubmit={[Function]}
    >
      <div
        className="input-group"
      >
        <input
          autoComplete="off"
          className="form-control"
          name="content"
          onChange={[Function]}
          placeholder="component:Chat.placeholder"
          value=""
        />
        <div
          className="input-group-append"
        >
          <button
            className="btn btn-primary"
            disabled={false}
            type="submit"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-paper-plane fa-w-16 "
              data-icon="paper-plane"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M476 3.2L12.5 270.6c-18.1 10.4-15.8 35.6 2.2 43.2L121 358.4l287.3-253.2c5.5-4.9 13.3 2.6 8.6 8.3L176 407v80.5c0 23.6 28.5 32.9 42.5 15.8L282 426l124.6 52.2c14.2 6 30.4-2.9 33-18.2l72-432C515 7.8 493.3-6.8 476 3.2z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;
